### ISSUE TITLE
Duplicate protection for drafteffects AddCards

### ISSUE THREAD

**Issue by tiera3** [Senryoku/Draftmancer#766]
Any chance of duplicate protection with the drafteffects AddCards feature? I am using it for Jumpstart deck generation from facecards with inbuilt variance.
Example:
`   "draft_effects": [{ "type": "AddCards", "count": 2, "cards": [
     "Erratic Visionary (JMP)", "Erratic Visionary (JMP)", "Erratic Visionary (JMP)", 
     "Rhystic Study (JMP)", "Read the Runes (JMP)", "Mystic Archaeologist (JMP)"]
    }]`
can add two copies of Erratic Visionary. The intention was to have a 50% chance of Erratic Visionary and one of the other three cards, or ~17% chance of each of the other combinations.

**Comment by Senryoku** [comment]
1217a81 adds a `duplicateProtection` property to the `AddCards` draft effect (enabled by default).
I also added support for card count, meaning you'll be able to write it slightly more succinctly:
```
"draft_effects": [
  { 
    "type": "AddCards", 
    "count": 2, 
    "cards": [ 
      "3 Erratic Visionary (JMP)", 
      "Rhystic Study (JMP)", 
      "Read the Runes (JMP)", 
      "Mystic Archaeologist (JMP)"
    ], 
    "duplicateProtection": true 
  }
]
```

**Comment by tiera3** [comment]
It is working well.

FYI - support for cardcount works for the above, but not for the following.
"draft_effects": [
  { 
    "type": "AddCards", 
    "cards": [ 
      "7 Swamp (J22) 106"
    ], 
  }
You only get one in your pool.

It also doesn't work for:
"draft_effects": [
  { 
    "type": "AddCards", 
    "count": 2, 
    "cards": [ 
      "7 Swamp (J22) 106"
    ], 
  }
You get an error
![image](https://github.com/user-attachments/assets/bd1b8284-149a-4424-9882-e079ec1e6ae6)

**Comment by Senryoku** [comment]
Right. 989d52d should fix this issue.

**Comment by Senryoku** [comment]
Good catch. Should be fixed by 989d52d2181175e31da899fd667e79b50ec21ff7 and 66ee2a108d9c27fd3ae5c6f2278ab18b448f28e5
